insecure/cmd/corrupted_builder.go

```diff
@@ -19,17 +23,33 @@ const CorruptNetworkPort = 4300
 // CorruptedNodeBuilder creates a general flow node builder with corrupt network.
 type CorruptedNodeBuilder struct {
 	*cmd.FlowNodeBuilder
+	TopicValidatorEnabled bool
 }
 
 func NewCorruptedNodeBuilder(role string) *CorruptedNodeBuilder {
 	return &CorruptedNodeBuilder{
-		FlowNodeBuilder: cmd.FlowNode(role),
+		FlowNodeBuilder:       cmd.FlowNode(role),
+		TopicValidatorEnabled: false,
 	}
 }
 
+// LoadCorruptFlags load additional flags for corrupt nodes.
+func (cnb *CorruptedNodeBuilder) LoadCorruptFlags() {
+	cnb.FlowNodeBuilder.ExtraFlags(func(flags *pflag.FlagSet) {
+		flags.BoolVar(&cnb.TopicValidatorEnabled, "topic-validator-enabled", false, "enable the libp2p topic validator for corrupt nodes")
+	})
+}
+
 func (cnb *CorruptedNodeBuilder) Initialize() error {
-	if err := cnb.FlowNodeBuilder.Initialize(); err != nil {
-		return fmt.Errorf("could not initilized flow node builder: %w", err)
+
+	// skip FlowNodeBuilder initialization if node role is access. This is because the AN builder uses
+	// a slightly different build flow than the other node roles. Flags and components are initialized
+	// in calls to anBuilder.ParseFlags & anBuilder.Initialize . Another call to FlowNodeBuilder.Initialize will
+	// end up calling BaseFlags() and causing a flags redefined error.
+	if cnb.NodeRole != flow.RoleAccess.String() {
```

Comment:
This will still need to call the AN version though right?

Ultimately, we should align the AN version to work more like the other nodes, but that's outside of the scope of this PR

Reply:
No is already called beforehand in the access node builder. The access node builder has a different flow than the EN and VN builders.